packages/data/lib/services/analytics_service_impl.dart

```diff
@@ -0,0 +1,15 @@
+import 'dart:async';
+
+import 'package:domain/services/analytics_service.dart';
+import 'package:firebase_analytics/firebase_analytics.dart';
+
+class AnalyticsServiceImpl implements AnalyticsService {
+  final FirebaseAnalytics _analyticsProvider;
+
+  AnalyticsServiceImpl(this._analyticsProvider);
+
+  @override
+  Future<void> logClickEvent(String eventName) async {
+    await _analyticsProvider.logEvent(name: eventName);
+  }
```

Comment:
in your case it's should be named logEvent, because u don't have something related to click in logging event